pre_commit_linter.py

```diff
@@ -0,0 +1,311 @@
+# coding: utf-8
+#
+# Copyright 2017 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Pre-commit script for oppia-ml-proto.
+
+This script lints Python code, and prints a
+list of lint errors to the terminal. If the directory path is passed,
+it will lint all Python files in that directory; otherwise,
+it will only lint files that have been touched in this commit.
+
+IMPORTANT NOTES:
+
+1.  Before running this script, you must install third-party dependencies by
+    running
+
+        bash install_prototool.sh
+
+    at least once.
+
+=====================
+CUSTOMIZATION OPTIONS
+=====================
+1.  To lint only files that have been touched in this commit
+        python pre_commit_linter.py
+
+2.  To lint all files in the folder or to lint just a specific file
+        python pre_commit_linter.py --path filepath
+
+3.  To lint a specific list of files (*.py only). Separate files by spaces
+        python pre_commit_linter.py --files file_1 file_2 ... file_n
+
+Note that the root folder MUST be named 'oppia-ml-proto'.
+"""
+
+# Pylint has issues with the import order of argparse.
+# pylint: disable=wrong-import-order
+import argparse
+import fnmatch
+import multiprocessing
+import os
+import subprocess
+import sys
+import time
+# pylint: enable=wrong-import-order
+
+_PARSER = argparse.ArgumentParser()
+_EXCLUSIVE_GROUP = _PARSER.add_mutually_exclusive_group()
+_EXCLUSIVE_GROUP.add_argument(
+    '--path',
+    help='path to the directory with files to be linted',
+    action='store')
+_EXCLUSIVE_GROUP.add_argument(
+    '--files',
+    nargs='+',
+    help='specific files to be linted. Space separated list',
+    action='store')
+
+if not os.getcwd().endswith('oppia-ml-proto'):
+    print('')
+    print('ERROR    Please run this script from the oppia root directory.')
+
+_PROTOTOOL_PATH = os.path.join(
+    os.getcwd(), 'third_party', 'prototool-1.10.0', 'prototool')
+
+if not os.path.exists(_PROTOTOOL_PATH):
+    print('')
+    print('ERROR    Please run install_prototool.sh first to install prototool')
+    sys.exit(1)
+
+_MESSAGE_TYPE_SUCCESS = 'SUCCESS'
+_MESSAGE_TYPE_FAILED = 'FAILED'
+
+EXCLUDED_PATHS = ('third_party/*',)
+
+PROTOTOOL_CONFIG_FILE = 'prototool_config.json'
+
+def _get_changed_filenames():
+    """Returns a list of modified files (both staged and unstaged)
+
+    Returns:
+        a list of filenames of modified files.
```

Comment:
Missing typeinfo. Start descriptions with capital letters. Ditto below.

Btw -- this is unrelated, but perhaps worth adding a python linter?

Reply:
I don't think that a python linter is necessary as the repo would contain only `.proto` files, except for linter and installer scripts. There won't be any third party usable python code in this repo.

Reply:
Done